domain/model/mesh_results.py

```diff
@@ -37,8 +41,14 @@ def get_alias(self, agent_id: AgentID) -> str:
             return f"[agent_id={agent_id} not found]"
         return agent.alias
 
-    def insert(self, agent: Agent) -> None:
-        self._agents[agent.id] = agent
+    def list_agent_ids(self) -> List[AgentID]:
+        return list(self._agents)
```

Comment:
Return `set`. Order does not matter, dict does not preserve it anyway and prevents duplicate IDs from being stored.